### PR TITLE
Upgrade Gradle to 7.3.3 & Android Gradle Plugin to 7.1.1

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR updates Gradle and Android Gradle Plugin to their latest versions. It builds on top of #2 and should be reviewed/merged after it.

* In 6dba9fbc5817bd1d46924245b4f45b6044922fbc, Gradle wrapper is updated with the `./gradlew wrapper --gradle-version 7.3.3 --distribution-type all` command.
* 059c3137fc50718235a97a5f1de9f3a762e90842 updates Android Gradle Plugin to its latest version `7.1.0`. I was a little reluctant to go to the latest version, but since there weren't any build errors specific to this version, I figured this way we'd test everything only once. I am open to suggestions if there is any preference for going with another version.
* d38059a77dc77c55cd4c10fc6381ec5cde24716b updates the compile & target sdk's to `30`. This is a required change for the Android Gradle Plugin update and would have been necessary even if we upgraded to `4.2.2`. (latest version before `7.0`) It also drops the `buildToolsVersion` which has been deprecated and no longer necessary.

I have some reservations about upgrading the target sdk due to my unfamiliarity with the project as the change needs to be tested thoroughly. I am hoping the reviewers can help me with this.

There is a good chance we'd want to test its integration to the Tumblr app before we approve this PR, which is completely fine by me. I'll keep opening PRs that build on top of each other and when we get to publishing the library artifacts, we can test the integration. This approach should still make it easier to review the PRs & follow along the changes. 🤞

### CODE REVIEW

File: sample/build.gradle [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 android {
-    compileSdkVersion 28
-    buildToolsVersion "30.0.2"
+    compileSdkVersion 30
 
     defaultConfig {
         applicationId "com.tumblr.permissme.sample"
         minSdkVersion 16
-        targetSdkVersion 28
+        targetSdkVersion 30
```
Comment (obsantos) [inline sample/build.gradle]: Should we target 31? We are working on upgrading this on the main repo, and the main blocker we have is the lack of Jdk 11 on Jenkins (which is also required for AGP 7.x), so we can do both at the same time

Comment (obsantos) [reply sample/build.gradle]: same for `compileSdkVersion`

**Review by obsantos — APPROVED** [review]
LGTM 👍

File: sample/build.gradle (hunk 1 above)
Comment (oguzkocer) [reply sample/build.gradle]: @obsantos Sounds good, addressed it in 3b617ac96a3f12c22a63d1c208293e5c46f6b75c.

Note that there was a spacing issue in the `AndroidManifest` file in my initial commit. The force push only updates that and doesn't change the previous commits. I didn't want to create a secondary commit just for that.

**Pull request merged**